### PR TITLE
Switch to 64-bit ints.

### PR DESCRIPTION
**Pull request by bruvzg**
Syncing features with https://github.com/godotengine/godot/pull/86730, but since no GDExtension public API is changed, it should be compatible both ways.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Should fix:
* https://github.com/godotengine/godot-cpp/issues/1338

**Review by dsnopek — APPROVED** [review]
Thanks!

This looks great to me :rocket: 

I `diff`'ed the `cowdata.hpp` and `vector.hpp` from this PR with the `cowdata.h` and `vector.h` from PR https://github.com/godotengine/godot/pull/86730, and they only had minor expected differences. Reading through all the rest, it all looks good.

I haven't tested it manually, but the fact that it's passing the automated tests is a very good sign.

**Comment by dsnopek** [comment]
I know we can merge this before Godot, however, I'm holding off on merging for a bit, just in case there's any further changes to `cowdata.h` and `vector.h` on the Godot PR that we'll want to sync over here. When that one seems like it's finished and about to be merged, we can merge here.

**Comment by dsnopek** [comment]
Thanks!

**Comment by dsnopek** [comment]
Cherry-picking to 4.1 in https://github.com/godotengine/godot-cpp/pull/1373

**Comment by AThousandShips** [comment]
Assuming this doesn't actually depend on the upstream changes? As they aren't cherry picked, we might want to be careful here as it does have some side effects that might be harmful:
* https://github.com/godotengine/godot/issues/87442

I'd hold off cherry picking this until we have worked things out

**Comment by bruvzg** [comment]
It doesn't depend on upstream change, but won't fix any of the issues without them, so is not useful on its own.

File: include/godot_cpp/templates/cowdata.hpp
Comment (AThousandShips) [inline include/godot_cpp/templates/cowdata.hpp:55]: This specifically breaks on some platforms, still being investigated

**Comment by YuriSizov** [comment]
> Should fix:
> 
> * [string length type inconsistency #1338](https://github.com/godotengine/godot-cpp/issues/1338)

Did it fix this issue btw?

**Comment by dsnopek** [comment]
> > Should fix:
> > 
> > * [string length type inconsistency #1338](https://github.com/godotengine/godot-cpp/issues/1338)
> 
> Did it fix this issue btw?

Yes it did! I've just closed that issue.

**Comment by dsnopek** [comment]
Cherry-picked for 4.2 in PR https://github.com/godotengine/godot-cpp/pull/1410

**Comment by dsnopek** [comment]
Cherry-picked for 4.1 in PR https://github.com/godotengine/godot-cpp/pull/1411

**Pull request merged**
